File: 1200/1909B/main.java
import java.util.*;

public class main
{
	public static void main(String[] args) 
	{
		Scanner sc = new Scanner(System.in);
 
		int t = sc.nextInt();
        while(t-- > 0)
        {
            int n = sc.nextInt();
			long a[] = new long[n];
			
			for(int i=0;i<n;i++)
				a[i] = sc.nextLong();
			
			long ans = 1;
			boolean flag = false;
			
			while(!flag)
			{
				ans = ans*2;
				HashSet<Long> mod = new HashSet<>();
				for(int i=0;i<n;i++)
					mod.add(a[i]%ans);
				
				if(mod.size() == 2)
					flag = true;
			}
			
			System.out.println(ans);
        }
	}
	
}

File: 1200/1909B/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;
    while (t--) 
    {
        int n;
        cin >> n;
        vector<long long> a(n);
        for (int i = 0; i < n; i++)
            cin >> a[i];

        long long ans = 1;
        bool flag = false;

        while (!flag) 
        {
            ans *= 2;
            unordered_set<long long> mod;

            for (int i = 0; i < n; i++)
                mod.insert(a[i] % ans);
            
            if (mod.size() == 2)
                flag = true;
        }

        cout << ans << "\n";
    }

    return 0;
}
